framework/src/org/apache/cordova/engine/SystemWebChromeClient.java

```diff
@@ -251,48 +251,40 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent) {
     }
 
-    // Validation utility for mime types
-    private List<String> extractValidMimeTypes(String[] mimeTypes) {
-        List<String> results = new ArrayList<String>();
-        List<String> mimes;
-        if (mimeTypes.length == 1 && mimeTypes[0].contains(",")) {
-            mimes = Arrays.asList(mimeTypes[0].split(","));
-        } else {
-            mimes = Arrays.asList(mimeTypes);
-        }
-        MimeTypeMap mtm = MimeTypeMap.getSingleton();
-        for (String mime : mimes) {
-            if (mime != null && mime.trim().startsWith(".")) {
-                String extensionWithoutDot = mime.trim().substring(1, mime.trim().length());
-                String derivedMime = mtm.getMimeTypeFromExtension(extensionWithoutDot);
-                if (derivedMime != null && !results.contains(derivedMime)) {
-                    // adds valid mime type derived from the file extension
-                    results.add(derivedMime);
-                }
-            } else if (mtm.getExtensionFromMimeType(mime) != null && !results.contains(mime)) {
-                // adds valid mime type checked agains file extensions mappings
-                results.add(mime);
-            }
-        }
-        return results;
-    }
-
     @TargetApi(Build.VERSION_CODES.LOLLIPOP)
     @Override
     public boolean onShowFileChooser(WebView webView, final ValueCallback<Uri[]> filePathsCallback, final WebChromeClient.FileChooserParams fileChooserParams) {
         Intent intent = fileChooserParams.createIntent();
         List<String> validMimeTypes = extractValidMimeTypes(fileChooserParams.getAcceptTypes());
-        if (validMimeTypes.isEmpty()) {
-            intent.setType(DEFAULT_MIME_TYPE);
-        } else {
-            intent.setType(String.join(" ", validMimeTypes));
+
+        // Update with mime types
+        intent.setType(DEFAULT_MIME_TYPE);
+
+        if (!validMimeTypes.isEmpty()) {
+            // Update with additional mime types here using a String[].
+            intent.putExtra(Intent.EXTRA_MIME_TYPES, validMimeTypes.toArray());
         }
+
+        // Only pick openable and local files. Theoretically we could pull files from google drive
+        // or other applications that have networked files, but that's unnecessary for this example.
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+
         try {
             parentEngine.cordova.startActivityForResult(new CordovaPlugin() {
                 @Override
                 public void onActivityResult(int requestCode, int resultCode, Intent intent) {
                     Uri[] result = WebChromeClient.FileChooserParams.parseResult(resultCode, intent);
-                    LOG.d(LOG_TAG, "Receive file chooser URL: " + result);
+
+                    String completeString = new String();
+                    ArrayList<String> fileList = new ArrayList<String>();
+                    for (Uri uri : result) {
+                        fileList.add(getFileName(uri));
```

Comment:
get real file name instead of reference to mimes:
![image](https://user-images.githubusercontent.com/2271337/66272517-fd93a580-e86a-11e9-812a-2cde5eb69325.png)